File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        map<char,int> mp;
        int ans=0;
        int i=0;
        int j=0;
        while(j<s.size()){
            if(mp.find(s[j])!=mp.end()){
                while(mp.find(s[j])!=mp.end()){
                    mp.erase(s[i]);
                    i++;
                }
            }
            mp[s[j]]++;
            ans=max(ans,j-i+1);
            j++;
        }
        return ans;
        
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Map<Character,Integer> mp=new HashMap<>();
        int i=0,j=0,ans=0;
        while(j<s.length()){
            if(mp.containsKey(s.charAt(j))){
                while(mp.containsKey(s.charAt(j))){
                    mp.remove(s.charAt(i));
                    i++;
                }
            }
            mp.put(s.charAt(j),mp.getOrDefault(s.charAt(j),0)+1);
            ans=Math.max(ans,j-i+1);
            j++;
        }
        return ans;
        
    }
}